Makefile

```diff
@@ -176,6 +176,13 @@ out/minikube_$(DEB_VERSION).deb: out/minikube-linux-amd64
 	dpkg-deb --build out/minikube_$(DEB_VERSION)
 	rm -rf out/minikube_$(DEB_VERSION)
 
+out/minikube_$(DEB_VERSION).tar.gz: out/localkube out/minikube-linux-amd64 out/minikube-darwin-amd64 out/minikube-windows-amd64.exe
+	rm -rf out/minikube_$(DEB_VERSION)
```

Comment:
I think we can simply this by just having a `TAR_TARGETS := ` top level makefile variable. Then you can make the rule depend on that and the tarball rule implementation just becomes `tar -cfv minikube.tar $(TAR_TARGETS)`

If we do the platform specific tarballs, you might have to do something similar to the `out/minikube-%-amd64` rules with `TAR_TARGETS_ALL`, `TAR_TARGETS_DARWIN`, etc.